### PR TITLE
Plugging mixin

### LINKED ISSUE #16: Add plugging mixin

### PR DESCRIPTION
**Pull request by ca-d**
Closes #16 and #69 . Nice.

### CODE REVIEW

File: open-scd.test.ts [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+beforeEach(() => {
+  editor = document.createElement('open-scd');
+  document.body.prepend(editor);
+});
+
+afterEach(() => {
+  editor.remove();
+});
+
+it(`changes locales on attribute change`, async () => {
```
Comment (JakobVogelsang) [inline open-scd.test.ts:36]: This is a test we already have in `visual.test.ts`. Is this duplication on purpose?

Comment (JakobVogelsang) [reply open-scd.test.ts:36]: Ahhh now I get it. You wanted to rename `visual.test.ts` to `open-scd.test.ts`. I think this is why the screenshots fail. You are creating new screenshots, but compare to the old once. Also, please remove the leftover `visual.test.ts`

**Review by JakobVogelsang — CHANGES_REQUESTED** [review]
There is a left over test file and I think some screenshot references are not valid anymore.

File: demo/index.html [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
-<open-scd></open-scd>
+<open-scd plugins='{"menu": [{"name": "Add plugins...", "translations": {"de": "Erweitern..."}, "icon": "extension", "active": true, "src": "/demo/AddPlugins.js"}]}'></open-scd>
 
 <script type="module">
 import '../dist/open-scd.js';
+
+const editor = document.querySelector('open-scd');
+const params = (new URL(document.location)).searchParams;
+for (const [name, value] of params) {
+console.log(`${name}: `, value, typeof value, value === '');
```
[the remark this answers is not in the document]
Comment (ca-d) [reply demo/index.html:12]: Yes it is, thanks for noticing :heavy_heart_exclamation:

File: open-scd.test.ts (hunk 1 above)
Comment (ca-d) [reply open-scd.test.ts:36]: Where is the leftover `visual.test.ts`? I can't find it for some reason.

Comment (ca-d) [reply open-scd.test.ts:36]: Looks like this was completely unrelated to the file rename and just had to do with lit-localize's `locales/de.js` not being picked up in the distribution directory. After moving back from `inputFiles` to `tsConfig` in `lit-localize.json`, localizations are now being emitted as typescript again (`locales-de.ts`) and therefore being included in `dist` by `tsc`, resolving the issue.

Comment (ca-d) [reply open-scd.test.ts:36]: What it says under https://lit.dev/docs/localization/cli-and-config/#general-settings is not true by the way, `inputFiles` does not supercede `tsConfig`, it's sadly the other way round. So you can't currently have custom input files while using TypeScript.

**Review by JakobVogelsang — APPROVED** [review]
LGTM

**Pull request merged**